src/Fixer/ListNotation/ListSyntaxFixer.php

```diff
@@ -0,0 +1,155 @@
+
+/**
+ * @author SpacePossum
+ */
+final class ListSyntaxFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface
+{
+    private $candidateTokenKind;
+    private $fixCallback;
+
+    /**
+     * Use 'syntax' => 'long'|'short'.
+     *
+     * @param array<string, string>|null $configuration
+     *
+     * @throws InvalidFixerConfigurationException
+     */
+    public function configure(array $configuration = null)
+    {
+        parent::configure($configuration);
+
+        $this->resolveCandidateTokenKind();
+        $this->resolveFixCallback();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'List (`array` destructuring) assignment should be declared using the configured syntax. Requires PHP >= 7.1.',
+            array(
+                new VersionSpecificCodeSample(
+                    "<?php\n[\$sample] = \$array;",
+                    new VersionSpecification(70100),
+                    array('syntax' => 'long')
+                ),
+                new VersionSpecificCodeSample(
+                    "<?php\nlist(\$sample) = \$array;",
+                    new VersionSpecification(70100),
+                    array('syntax' => 'short')
+                ),
+            )
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // should be run before the BinaryOperatorSpacesFixer and TernaryOperatorSpacesFixer.
+        return 1;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return PHP_VERSION_ID >= 70100 && $tokens->isTokenKindFound($this->candidateTokenKind);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $callback = $this->fixCallback;
+        for ($index = $tokens->count() - 1; 0 <= $index; --$index) {
+            if ($tokens[$index]->isGivenKind($this->candidateTokenKind)) {
+                $callback($tokens, $index);
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        $syntax = new FixerOptionBuilder('syntax', 'Whether to use the `long` or `short` `list` syntax.');
+        $syntax = $syntax
+            ->setAllowedValues(array('long', 'short'))
+            ->setDefault('long')
+            ->getOption()
+        ;
+
+        return new FixerConfigurationResolver(array($syntax));
+    }
+
+    private function resolveCandidateTokenKind()
+    {
+        $this->candidateTokenKind = 'long' === $this->configuration['syntax']
+            ? CT::T_DESTRUCTURING_SQUARE_BRACE_OPEN
+            : T_LIST
+        ;
+    }
+
+    private function resolveFixCallback()
+    {
+        if ('long' === $this->configuration['syntax']) {
+            $this->fixCallback = function (Tokens $tokens, $index) {
```

Comment:
basically not having to do the `if` condition every time in the loop while the result is always the same